api/routes/tasks.py

```diff
@@ -79,4 +131,15 @@ def delete_task(user_id):
     # body has task id
     body = request.get_json()
 
-    db.delete(Task).where(Task.user_id == user_id and Task.id == body["id"])
+    task = (
+        db.session.query(Task)
+        .where(Task.user_id == user_id, Task.task_id == body["id"])
+        .one()
+    )
+    if task is None:
+        abort(400)
```

Comment:
Would `404` be a more appropriate status code in this case, as this is the branch we fall into when we fail to find a task with that id.